bot.py

```diff
@@ -296,6 +300,31 @@ async def create_or_get_voice_client(self, message):
 
         return self.voice_client
 
+    async def attempt_disconnect(self):
+        """
+        Should be called whenever a song finishes playing, or when media is paused or
+        stopped. Attempts to disconnect the voice client after a set amount of time by
+        checking whether anything is currently playing.
+        """
+        logging.info(
+            "Will attempt to disconnect in %s seconds",
+            MusicBot.DISCONNECT_TIMER_SECONDS,
+        )
+        self.last_played_time = time()
+        await asyncio.sleep(MusicBot.DISCONNECT_TIMER_SECONDS)
+
+        if self.voice_client.is_playing():
+            return
+
+        buffer = 5
+        time_since_last_played = time() - self.last_played_time
+        if time_since_last_played + buffer < self.DISCONNECT_TIMER_SECONDS:
```

Comment:
```suggestion
        if abs(time_since_last_played - self.DISCONNECT_TIMER_SECONDS) < buffer :
```
Allows timer drift in either direction

Reply:
Can this even happen?

Reply:
That's a good point gardarandri. Fixed.